src/components/PopupWithForm.js

```diff
@@ -32,12 +34,15 @@ export default class PopupWithForm extends Popup {
     super.removeEventListeners();
   }
 
-  _handleSubmitForm = () => {
+  _handleSubmitForm = (evt) => {
+    this._submitBtn.textContent = 'Saving...';
```

Comment:
Needs correction: You should bot start the loading here, since in not all the submission cases the loading state can be needed. Please do that in index.js.

Also you should return the initial button text after loading.

```js
constructor(popupSelector, { handleSubmit, loadingButtonText }) { 
  …
  // saving the initial button text
  this._buttonText = this._submitBtn.textContent;
  this._loadingButtonText = loadingButtonText;
}

// call this method in index.js before api call to start the loading and in `finally` block to end the loading
renderLoading(isLoading) {
    if (isLoading) { 
      this._submitButton.textContent = this._loadingButtonText; 
    } else { 
      this._submitButton.textContent = this._buttonText; 
    } 
}
```